### PR TITLE
Prefer `Display` over `Debug`

### LINKED ISSUE #16104: Consistent formatting of entities in error messages
## How can Bevy's documentation be improved?

Some error messages (such as in `EntityCommands::insert`) `Debug`-format entities, while others (such as in `InsertReflect`) use `Display`-formatting. This should be made more consistent.

Display format is `{id}v{generation}` while Debug format is `{id}v{generation}#{raw bits}`. The raw bits contain the same data, but are useful for comparing with serialized entities.

### PR DESCRIPTION
**Pull request by BenjaminBrienen**
# Objective

Fixes #16104

## Solution

I removed all instances of `:?` and put them back one by one where it caused an error.

I removed some bevy_utils helper functions that were only used in 2 places and don't add value. See: #11478

## Testing

CI should catch the mistakes

## Migration Guide

`bevy::utils::{dbg,info,warn,error}` were removed. Use `bevy::utils::tracing::{debug,info,warn,error}` instead.

### CODE REVIEW

**Comment by BenjaminBrienen** [comment]
@SpecificProtagonist feel free to review or point out good places to use `disqualified::ShortName` here.

**Comment by mockersf** [comment]
Why is there a migration guide? There's no information in the description about this change.

I would prefer this to be blocked until after the 0.15 is released, as it has high chances of conflicts during cherry picks.

**Comment by mockersf** [comment]
Controversial by extension of #16113 as it was used to do this PR

**Comment by bushrat011899** [comment]
I agree with the changes made here, but I don't think it's worth it unless we have some lints to enforce the new standard. If we had `format_debug_instead_of_display` and `format_parameter_instead_of_inline` (made-up names) then we'd be in a better position to make this change long-term. Right now, this change will be made and then almost immediately regressed, since we have neither tooling nor policy to enforce it.

**Comment by BenjaminBrienen** [comment]
> I agree with the changes made here, but I don't think it's worth it unless we have some lints to enforce the new standard. If we had `format_debug_instead_of_display` and `format_parameter_instead_of_inline` (made-up names) then we'd be in a better position to make this change long-term. Right now, this change will be made and then almost immediately regressed, since we have neither tooling nor policy to enforce it.

There is https://rust-lang.github.io/rust-clippy/master/index.html?levels=allow%2Cwarn%2Cnone&groups=nursery%2Cpedantic%2Crestriction%2Cstyle#uninlined_format_args

**Comment by BenjaminBrienen** [comment]
I refined this PR to not have the formatting changes

**Review by bushrat011899 — APPROVED** [review]
I think this is good, but as stated I would open an issue for a lint which keeps this change maintained long-term.

File: crates/bevy_ecs/src/system/mod.rs [hunk 1]
```diff
@@ -1642,7 +1642,13 @@ mod tests {
...
         assert_is_system(returning::<&str>.map(u64::from_str).map(Result::unwrap));
         assert_is_system(static_system_param);
-        assert_is_system(exclusive_in_out::<(), Result<(), std::io::Error>>.map(bevy_utils::error));
+        assert_is_system(
+            exclusive_in_out::<(), Result<(), std::io::Error>>.map(|system| {
+                if let Err(err) = system {
+                    bevy_utils::tracing::error!("{err}");
+                }
+            }),
+        );
```
[the remark this answers is not in the document]
Comment (BenjaminBrienen) [reply crates/bevy_ecs/src/system/mod.rs:1651]: I'm confused, what feedback are you giving here?

File: crates/bevy_utils/src/lib.rs [hunk 2]
```diff
@@ -383,30 +383,6 @@ impl<F: FnOnce()> Drop for OnDrop<F> {
...
-    }
-}
-
-/// Processes a [`Result`] by calling the [`tracing::error!`] macro in case of an [`Err`] value.
-pub fn error<E: Debug>(result: Result<(), E>) {
-    if let Err(error) = result {
-        tracing::error!("{:?}", error);
-    }
-}
-
```
[the remark this answers is not in the document]
Comment (BenjaminBrienen) [reply crates/bevy_utils/src/lib.rs:409]: There is a goal to shrink utils as much as possible and these are not really used within bevy.

[the remark this answers is not in the document]
Comment (mnmaita) [reply crates/bevy_utils/src/lib.rs:409]: Related, just to back the removal of these: https://github.com/bevyengine/bevy/pull/13556#issuecomment-2135968940

File: crates/bevy_ui/src/layout/mod.rs [hunk 3]
```diff
@@ -177,7 +177,7 @@ pub fn ui_layout_system(
                 Some(camera_entity) => {
                     let Ok((_, camera)) = cameras.get(camera_entity) else {
                         warn!(
-                            "TargetCamera (of root UI node {entity:?}) is pointing to a camera {:?} which doesn't exist",
+                            "TargetCamera (of root UI node {entity:?}) is pointing to a camera {} which doesn't exist",
```
Comment (mnmaita) [inline crates/bevy_ui/src/layout/mod.rs:180]:
```suggestion
                            "TargetCamera (of root UI node {entity}) is pointing to a camera {} which doesn't exist",
```
I guess there's a leftover `:?` there, right?

File: crates/bevy_ecs/src/system/mod.rs [hunk 4]
```diff
@@ -1642,7 +1642,13 @@ mod tests {
         assert_is_system(returning::<Option<()>>.map(drop));
         assert_is_system(returning::<&str>.map(u64::from_str).map(Result::unwrap));
         assert_is_system(static_system_param);
-        assert_is_system(exclusive_in_out::<(), Result<(), std::io::Error>>.map(bevy_utils::error));
+        assert_is_system(
+            exclusive_in_out::<(), Result<(), std::io::Error>>.map(|system| {
+                if let Err(err) = system {
```
Comment (mnmaita) [inline crates/bevy_ecs/src/system/mod.rs:1647]:
```suggestion
            exclusive_in_out::<(), Result<(), std::io::Error>>.map(|out| {
                if let Err(err) = out {
```

Just to keep it consistent with a previous change request.

**Review by mnmaita — APPROVED** [review]
Nice one, leaving two minor suggestions.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@BenjaminBrienen CI failures are real; I think you need to revert a couple of things here.

**Pull request merged**